### PR TITLE
add i8l for promo code redemption

### PR DESCRIPTION
**Pull request by jackewing**
ADO card: ☑️ [AB#6181](https://dev.azure.com/S72/SHIFT72/_boards/board/t/Web%20team/Stories/?workitem=6181)

Elab notes/AC: 📃 [redeem content](https://docs.google.com/document/d/13WbZQ7XgpTpGfIYYhggcXDFE5TOuFjl1yYMRj2nZ_zI/edit?pli=1#)
- [x] Has this been discussed with Delivery Team?

Designs: 
https://docs.google.com/document/d/13WbZQ7XgpTpGfIYYhggcXDFE5TOuFjl1yYMRj2nZ_zI/edit?pli=1#

## Description of work
Adds translation strings for promo code content redemption.

### Any PRs dependent on this one
https://github.com/indiereign/shift72-web/pull/316/files

### Affected Clients
 - All clients who use Kibble

## Checklist
- [x] CI tests are passing Github actions (inc. linting)
- [x] Key areas of the feature outlined for context and testing
- [x] If there are designs for this work are they noted here and in the ADO card 
- [x] Design review
- [x] Have checked this at multiple screen resolutions and range of browsers
- [x] Moved ADO card to Dev/done, checked link to Github, tagged "Review"
- [x] Updated changelog (if applicable)
- [x] I promise to document any new feature toggles/configurations in the appropriate documentation

### CODE REVIEW

**Comment by PeculiarGoat** [comment]
Looks like you missed translations for 25 languages there :stuck_out_tongue:

**Comment by jackewing** [comment]
Though you wouldn't notice. I tried out @additiverich's translation script. Seemed to work a treat. I might've gotten overzealous in removing trailing spaces between interpolations and full stops.

File: site/zh_TW.all.json [hunk 1]
```diff
@@ -1319,5 +1319,20 @@
...
     "other": "更新您的信用卡。支持的卡是 Visa、Mastercard 和 American Express。"
+  },
+  "shopping_discount_applied_promo": {
+    "other": "代碼成功應用。"
+  },
+  "shopping_error_promo_only_purchases_must_be_100_percent_discounted": {
+    "other": "請輸入有效的促銷代碼。"
+  },
+  "shopping_complete_promo_only": {
+    "other": "成功！您現在可以觀看{{.Title}} 。"
```
Comment (PeculiarGoat) [inline site/zh_TW.all.json:1330]: Is that space supposed to be there before the dot?

File: site/ar_LB.all.json [hunk 2]
```diff
@@ -1338,5 +1338,20 @@
   },
   "shopping_card_update_reason_expired": {
     "other": "قم بتحديث بطاقتك الائتمانية. البطاقات المدعومة هي Visa و Mastercard و American Express."
+  },
+  "shopping_discount_applied_promo": {
+    "other": "تم تطبيق الكود بنجاح."
+  },
+  "shopping_error_promo_only_purchases_must_be_100_percent_discounted": {
```
Comment (PeculiarGoat) [inline site/ar_LB.all.json:1345]: Can we please _not_ have these nightmarish error code concatenated translation keys?  They are a pain to maintain:

The `promo_only_purchases_must_be_100_percent_discounted` string doesn't exist in core-template or Relish so to find the usage I have to search for `shopping_error_` and guess which of the 5 results actually wants to use this translation.

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]

File: site/ar_LB.all.json (hunk 2 above)
Comment (jackewing) [reply site/ar_LB.all.json:1345]: It arrives in the attached relish pr that's in review. Would you prefer it to be shorter or follow a different naming scheme?

File: site/zh_TW.all.json (hunk 1 above)
Comment (jackewing) [reply site/zh_TW.all.json]: Removed. TIL this is a different character than a period, it must've escaped my find and replace

File: site/ar_LB.all.json (hunk 2 above)
Comment (PeculiarGoat) [reply site/ar_LB.all.json:1345]: I would prefer the keys to be hard coded in Relish [like this](https://github.com/indiereign/shift72-web/blob/ee6eb5061a8f6a136fb624fb8ceb031fb3e63b74/src/transactional/shopping/shopping.service.js#L10).

Comment (jackewing) [reply site/ar_LB.all.json:1345]: Have hardcoded the error in the associated relish per https://github.com/indiereign/shift72-web/pull/316/files

Comment (PeculiarGoat) [reply site/ar_LB.all.json:1345]: Nice.  You're gonna need to remove `shopping_error_` from all these keys then right?

Comment (jackewing) [reply site/ar_LB.all.json]: Done

**Review by PeculiarGoat — APPROVED** [review]
Nice 👍

**Pull request merged**